app/src/main/java/de/westnordost/streetcomplete/map/MainFragment.kt

```diff
@@ -183,21 +183,17 @@ class MainFragment : Fragment(R.layout.fragment_main),
 
     override fun onConfigurationChanged(newConfig: Configuration) {
         super.onConfigurationChanged(newConfig)
-        val mapFragment = mapFragment ?: return
-        /* when rotating the screen and the bottom sheet is open, the view should not rotate around
-           its proper center but around the center of the part of the map that is not occluded by
-           the bottom sheet */
-        if (bottomSheetFragment != null) {
-            val currentPos = mapFragment.getViewPosition(mapOffsetWithOpenBottomSheet)
+        mapFragment?.let { map ->
+            val previousOffset = mapOffsetWithOpenBottomSheet
             updateMapQuestOffsets()
-            if (currentPos != null) {
-                val offsetPos = mapFragment.getPositionThatCentersPosition(currentPos, mapOffsetWithOpenBottomSheet)
-                mapFragment.updateCameraPosition { position = offsetPos }
+            /* when rotating the screen and the bottom sheet is open, the view
+               should not rotate around its proper center but around the center
+               of the part of the map that is not occluded by the bottom sheet */
+            if (bottomSheetFragment != null) {
+                map.adjustToOffsets(previousOffset, mapOffsetWithOpenBottomSheet)
             }
-        } else {
-            updateMapQuestOffsets()
```

Comment:
Trade-off here:

- Before, we called `updateMapQuestOffsets` in both branches of the check if the bottomSheetFragment is open. I didn't like this; I'd prefer to see at a glance that the offset is always updated.
- Now, I moved it outside the check. However, this means we always need to save the previous offset position, even though it's only used if the bottom sheet is open.

Overall I think this is worth it, but just wanted to point out it's a place where you might disagree.

Reply:
Also, I moved the line breaks in the comment so it stays under 80 characters per line (from 100).
In general, I think that 100 makes sense, but when I can stay at 80 characters without making other sacrifices, I prefer that. In this case, the comment takes 3 lines, either way.

Reply:
Sure, looks good! Though, while we are at cosmetics, I generally prefer the pattern
```kotlin
val mapFragment = mapFragment ?: return
```
over
```kotlin
mapFragment?.let { mapFragment ->... }
```
because one level of indentation is saved this way. (Also, imagine you have multiple such properties, then the second one becomes a christmas tree)

Reply:
> I generally prefer the pattern
> 
> ```kotlin
> val mapFragment = mapFragment ?: return
> ```

I generally prefer functional styles — the open `return` here makes me uncomfortable. So if I were going to write it with a `let`, I'd write,

```kotlin
let mapFragment = this.mapFragment;
if (mapFragment != null) {
   /* ... */
}
```

…at which point `let` is just syntactic sugar to save a line. I still do use this form when there's an `else` branch, because this looks ugly to me

```kotlin
this.mapFragment?.let { mapFragment ->
    /* ... */
} ?: {
    /* ... */
}
```
If I could use `else` instead of `?:` there, I probably would use that form, though ;)

Unnecessary shadowing also makes me uncomfortable, so I'd either use a different variable name or disambiguate with `this`.

---

But, I do acknowledge this is mostly bikeshedding, so I'm happy to change the style so it's consistent with the rest of the code.

Reply:
> Unnecessary shadowing also makes me uncomfortable, so I'd either use a different variable name or disambiguate with this.

I'd be fine with this too

Reply:
okay, I went with `this.`, since it's more in line with the other locations where we do this.